src/main/Commands.cpp

```diff
@@ -0,0 +1,921 @@
+    }
+
+    auto& cfg = *options.mConfig;
+    VirtualClock clock(VirtualClock::REAL_TIME);
+    cfg.setNoListen();
+    auto app = Application::create(clock, cfg, false);
+
+    Json::Value catchupInfo;
+    auto result = catchup(app, parseCatchup(catchupString), catchupInfo);
+    if (!catchupInfo.isNull())
+        writeCatchupInfo(catchupInfo, outputFile);
+
+    return result;
+}
+
+int
+runCheckQuorum(Args const& args)
+{
+    auto options = handleConfigurationOptions(args, Parser{},
+                                              []() { return std::string{}; });
+    if (options.hasExitCode())
+    {
+        return options.exitCode();
+    }
+
+    checkQuorumIntersection(*options.mConfig);
+    return 0;
+}
+
+int
+runConvertId(Args const& args)
+{
+    std::string id;
+    auto parser = Parser{} | Arg(id, "ID").required();
+    auto options = handleHelpOption(args, parser, [&]() {
+        if (id.empty())
+        {
+            return std::string{"ID argument is required"};
+        }
+        else
+        {
+            return std::string{};
+        }
+    });
+    if (options.hasExitCode())
+    {
+        return options.exitCode();
+    }
+
+    StrKeyUtils::logKey(std::cout, id);
+    return 0;
+}
+
+int
+runDumpXDR(Args const& args)
+{
+    std::string xdr;
+    auto parser = Parser{} | Arg(xdr, "XDR").required();
+    auto options = handleHelpOption(args, parser, [&]() {
+        if (xdr.empty())
+        {
+            return std::string{"XDR argument is required"};
+        }
+        else
+        {
+            return std::string{};
+        }
+    });
+    if (options.hasExitCode())
+    {
+        return options.exitCode();
+    }
+
+    dumpxdr(xdr);
+    return 0;
+}
+
+int
+runForceSCP(Args const& args)
+{
+    bool set = true;
+    auto parser =
+        Parser{} |
+        Opt(set, "FLAG")["--set"](
+            "next time stellar-core is run, SCP will start with the local "
+            "ledger rather than waiting to hear from the network");
+    auto options = handleConfigurationOptions(args, parser,
+                                              []() { return std::string{}; });
+    if (options.hasExitCode())
+    {
+        return options.exitCode();
+    }
+
+    setForceSCPFlag(*options.mConfig, set);
+    return 0;
+}
+
+int
+runFuzz(Args const& args)
+{
+    std::string fileName;
+    auto parser = Parser{} | Arg(fileName, "FILE").required();
+    auto options = handleCommonOptions(args, parser, [&]() {
+        if (fileName.empty())
+        {
+            return std::string{"FILE argument is required"};
+        }
+        else
+        {
+            return std::string{};
+        }
+    });
+    if (options.hasExitCode())
+    {
+        return options.exitCode();
+    }
+
+    fuzz(fileName, options.mLogLevel, options.mMetrics);
+    return 0;
+}
+
+int
+runGenFuzz(Args const& args)
+{
+    std::string fileName;
+    auto parser = Parser{} | Arg(fileName, "FILE").required();
```

Comment:
`FILE` -> `FILE-NAME`

also, maybe you should have a helper function for the predicate: it's quite a bit of copy/paste